Microsoft.Research/Contracts/System.Windows.Forms/System.Windows.Forms.DataGridViewCell.cs

```diff
@@ -0,0 +1,1265 @@
+        public virtual DataGridViewCellStyle GetInheritedStyle(DataGridViewCellStyle inheritedCellStyle, int rowIndex,
+            bool includeColors)
+        {
+            Contract.Requires(this.DataGridView != null);
+            Contract.Requires(rowIndex >= 0 || rowIndex < this.DataGridView.Rows.Count);
+            Contract.Requires(this.ColumnIndex < 0);
+            return default(DataGridViewCellStyle);
+        }
+
+        /// <summary>
+        /// Calculates the preferred size, in pixels, of the cell.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// A <see cref="T:System.Drawing.Size"/> that represents the preferred size, in pixels, of the cell.
+        /// </returns>
+        /// <param name="graphics">The <see cref="T:System.Drawing.Graphics"/> used to draw the cell.</param><param name="cellStyle">A <see cref="T:System.Windows.Forms.DataGridViewCellStyle"/> that represents the style of the cell.</param><param name="rowIndex">The zero-based row index of the cell.</param><param name="constraintSize">The cell's maximum allowable size.</param>
+        // protected virtual Size GetPreferredSize(Graphics graphics, DataGridViewCellStyle cellStyle, int rowIndex, Size constraintSize)
+        
+        /// <summary>
+        /// Gets the size of the cell.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// A <see cref="T:System.Drawing.Size"/> representing the cell's dimensions.
+        /// </returns>
+        /// <param name="rowIndex">The index of the cell's parent row.</param><exception cref="T:System.InvalidOperationException"><paramref name="rowIndex"/> is -1</exception>
+        protected virtual Size GetSize(int rowIndex)
+        {
+            Contract.Requires(rowIndex != -1);
+            return default(Size);
+        }
+        
+        /// <summary>
+        /// Gets the value of the cell.
+        /// </summary>
+        /// 
+        /// <returns>
+        /// The value contained in the <see cref="T:System.Windows.Forms.DataGridViewCell"/>.
+        /// </returns>
+        /// <param name="rowIndex">The index of the cell's parent row.</param><exception cref="T:System.ArgumentOutOfRangeException">The value of the <see cref="P:System.Windows.Forms.DataGridViewElement.DataGridView"/> property of the cell is not null and <paramref name="rowIndex"/> is less than 0 or greater than or equal to the number of rows in the parent <see cref="T:System.Windows.Forms.DataGridView"/>.</exception><exception cref="T:System.InvalidOperationException">The value of the <see cref="P:System.Windows.Forms.DataGridViewElement.DataGridView"/> property of the cell is not null and the value of the <see cref="P:System.Windows.Forms.DataGridViewCell.ColumnIndex"/> property is less than 0, indicating that the cell is a row header cell.</exception>
+        protected virtual object GetValue(int rowIndex)
+        {
+            Contract.Requires(this.DataGridView != null);
```

Comment:
OK, I'm giving up. The same chagnes should be made for all preconditions for other methods in this class...